common/buildcraft/lib/net/MessageUpdateTile.java

```diff
@@ -46,6 +49,7 @@ public void fromBytes(ByteBuf buf) {
     public void toBytes(ByteBuf buf) {
         PacketBuffer buffer = new PacketBuffer(buf);
         buffer.writeBlockPos(pos);
+        buffer.writeString(className);
```

Comment:
We do strings need to be written out? They are expensive, and relying on the correct class type seems... odd. Did you actually crash ever because a tile entity changed types too quickly?